src/main/java/org/spongepowered/common/mixin/core/item/inventory/MixinContainer.java

```diff
@@ -135,6 +138,18 @@ public void onPutStackInSlot(int slotId, ItemStack itemstack, CallbackInfo ci) {
         }
     }
 
+    @Inject(method = "slotClick", at = @At(value = "HEAD"), cancellable = true)
+    public void onSlotClick(int slotId, int clickedButton, int mode, EntityPlayer player, CallbackInfoReturnable<net.minecraft.item.ItemStack> ci) {
+        if (slotId >= inventorySlots.size()) {
+            // Kick the player who's sending invalid packets and cancel the click
```

Comment:
If it's kicking the player, then it might be worth considering this as an exploit of sorts.